File: TrelloApp/src/app/core/models/label.ts
import {Board} from './board';
import {CardLabel} from './card-label';

export interface Label {
  id: number,
  title: string,
  color: string,
  boardId: number,
  createdAt: string,
  updatedAt?: string,
  board: Board,
  cardLabels: CardLabel[],
}

export interface LabelWithAssignment extends Label {
  isAssigned: boolean;
}

export interface AddLabel {
  title: string;
  color: string;
}

export interface UpdateLabel {
  title?: string;
  color?: string;
}

File: TrelloApi/TrelloApi/src/Domain/Entities/Label.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("Label")]
public class Label: Entity
{
    [StringLength(32), Required] 
    public string Title { get; set; }

    [StringLength(8), Required] 
    public string Color { get; set; }

    [ForeignKey("Board"), Required]
    public int BoardId { get; set; }
    public Board Board { get; set; }

    public ICollection<CardLabel> CardLabels { get; set; }

    public Label(string title, string color, int boardId)
    {
        Title = title;
        Color = color;
        BoardId = boardId;
    }
}